src/providers/paypro/paypro.spec.ts

```diff
@@ -0,0 +1,115 @@
+import { TestUtils } from '../../test';
+import { ProfileProvider } from '../profile/profile';
+import { PayproProvider } from './paypro';
+
+describe('PayProProvider', () => {
+  let payproProvider: PayproProvider;
+  let profileProvider: ProfileProvider;
+  const walletFixture = {
+    id: {
+      credentials: {
+        coin: 'btc',
+        network: 'livenet'
+      },
+      isComplete: () => {
+        return true;
+      },
+      fetchPayPro: (_payProUrl, _cb) => {}
+    }
+  };
+
+  const walletFixture2 = {
+    id: {
+      credentials: {
+        coin: 'btc',
+        network: 'testnet'
+      },
+      isComplete: () => {
+        return true;
+      },
+      fetchPayPro: (_payProUrl, _cb) => {}
+    }
+  };
+
+  beforeEach(() => {
+    const testBed = TestUtils.configureProviderTestingModule();
+    payproProvider = testBed.get(PayproProvider);
+    profileProvider = testBed.get(ProfileProvider);
+  });
+
+  describe('getPayProDetails', () => {
+    it('should return paypro details', () => {
+      walletFixture.id.fetchPayPro = (_payProUrl, cb) => {
+        const paypro = {
+          verified: true
+        };
+        return cb(null, paypro);
+      };
+      profileProvider.wallet = walletFixture;
+      payproProvider
+        .getPayProDetails('https://bitpay.com/i/BAikNA9QFMFiEs8HtNnQVt', 'btc')
```

Comment:
This should be local data. No external connection need to be realize with Unit Testing.